src/frontend/src/flows/addDevice/manage/pollForTentativeDevice.ts

```diff
@@ -0,0 +1,134 @@
+import { html, render } from "lit-html";
+import { IIConnection } from "../../../utils/iiConnection";
+import { renderManage } from "../../manage";
+import { withLoader } from "../../../components/loader";
+import { verifyDevice } from "./verifyTentativeDevice";
+import {
+  Countdown,
+  formatRemainingTime,
+  setupCountdown,
+} from "../../../utils/countdown";
+import {
+  DeviceData,
+  IdentityAnchorInfo,
+} from "../../../../generated/internet_identity_types";
+
+const pageContent = (userNumber: bigint, endTimestamp: bigint) => html`
+  <div class="container">
+    <h1>Add New Remote Device</h1>
+    <p>
+      Device registration mode enabled for Identity Anchor
+      <b>${userNumber}</b>. Please follow these steps to add your new device:
+    </p>
+    <ol class="instruction-steps">
+      <li>
+        On your <i>new</i> device:<br />
+        Open <b>https://identity.ic0.app</b>
+      </li>
+      <li>
+        On your <i>new</i> device:<br />
+        Chose
+        <b>Already have an anchor but using a new device?</b>
+      </li>
+      <li>
+        On your <i>new</i> device:<br />
+        Enter your Identity Anchor
+        <b>${userNumber}</b>
+      </li>
+      <li>
+        On your <i>new</i> device:<br />
+        Choose an alias for your new device
+      </li>
+    </ol>
+    <p>
+      This page will automatically refresh after completing the above steps.
+    </p>
+    <p>
+      Time remaining:
+      <span id="timer">${formatRemainingTime(endTimestamp)}</span>
+    </p>
+    <button id="cancelAddRemoteDevice" class="linkStyle">Cancel</button>
+  </div>
+`;
+
+export const pollForTentativeDevice = async (
+  userNumber: bigint,
+  connection: IIConnection
+): Promise<void> => {
+  await withLoader(async () => {
+    const [timestamp, userInfo] = await Promise.all([
```

Comment:
Note: some funky stuff would happen if the browser's clock is slower than the canister's. Since you're anyway polling the canister every 2 seconds, why not return how much time is left and use that as the displayed value (+ some less reliable countdown in between requests) ? Then you'd be sure there'd be no drift at all